### PR TITLE
File Share Create & Update API Refactoring for Sync DB using go routine

### LINKED ISSUE #1052: Request TimeOut for FileShare APIs
**Issue/Feature Description:** File Share Operation is successful but API results request timeout.

Server throws an error.
`{"id":"go.micro.client","code":408,"detail":"context deadline exceeded","status":"Request Timeout"}`
image

**Why this issue to fixed / feature is needed(give scenarios or use cases):** To provide clean FileShare API operations

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by himanshuvar**
**What type of PR is this?**
> /kind enhancement

**What this PR does / why we need it**:
This PR is to fix Timeout Issues for Create and Update File Share API 

**Which issue(s) this PR fixes**:

Fixes #1052 

**Test Report Added?**:
> /kind TESTED

**Test Report**:

The following testing is done for DELETE File Share API:

- [x] Register Backend

- [x] List Backend

- [x] Create FileShare for AWS Backend

- [x] Create FileShare for Azure Backend

- [x] List all File Share

- [x] Get FileShare

- [x] Update FileShare for AWS Backend

- [x] Update FileShare for Azure Backend

- [x] Delete FileShare

**Register a Backend**

Example 1: Registering an Azure Backend for File Share

POST http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/backends

Request:
```
{
    "Name": "azure-backend-file",
    "Type": "azure-file",
    "Region": "East Asia",
    "Access": "zsftest",
    "Security": "wkiGL5gv6ljye6LK7aNG4KTkFAo1lJ2HRU0oAuAYXSIqAWRQpkcW4YRxaj2PZKPFAJPga/Z+Z0oeducxONll7Q=="
}
```
Response:
```
{
    "id": "5f21624864aeb90001d5bd3c",
    "tenantId": "94b280022d0c4401bcf3b0ea85870519",
    "userId": "558057c4256545bd8a307c37464003c9",
    "name": "azure-backend-file",
    "type": "azure-file",
    "region": "East Asia",
    "access": "myaccess",
    "security": "mysecret"
}
```

Example 2: Registering an AWS Backend for File Share

POST http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/backends
Request:
```
{
  "Name": "aws-backend-file",
  "Type": "aws-file",
  "Region": "ap-south-1",
  "Access": "myaccess",
  "Security": "mysecret"
}
```
Response:
```
{
    "id": "5f2162cc64aeb90001d5bd3d",
    "tenantId": "94b280022d0c4401bcf3b0ea85870519",
    "userId": "558057c4256545bd8a307c37464003c9",
    "name": "aws-backend-file",
    "type": "aws-file",
    "region": "ap-south-1",
    "access": "myaccess",
    "security": "mysecret"
}
```
**List Backend:**
GET http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/backends
Response:
```
{
    "backends": [
        {
            "id": "5f21624864aeb90001d5bd3c",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "name": "azure-backend-file",
            "type": "azure-file",
            "region": "East Asia"
        },
        {
            "id": "5f2162cc64aeb90001d5bd3d",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "name": "aws-backend-file",
            "type": "aws-file",
            "region": "ap-south-1"
        }
    ],
    "next": 2
}
```

**Create a File share**

Example 1: Creating File Share for AWS Backend

POST http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares
Request:
```
{
    "name": "ashitfs",
    "description": "AWS FileShare",
    "backendId": "5f2162cc64aeb90001d5bd3d",
    "availabilityZone": "ap-south-1a",
    "type": "cloudFS",
    "tags": [
      {
        "key": "Name",
        "value": "ashitfs"
      }
    ],
    "metadata": {
      "PerformanceMode": "generalPurpose",
      "ThroughputMode": "bursting"
    }
}
```
Response:
```
{
    "id": "5f21646ecb68c700018b8625",
    "createdAt": "2020-07-29T17:28:38",
    "updatedAt": "2020-07-29T17:28:38",
    "name": "ashitfs",
    "description": "AWS FileShare",
    "tenantId": "94b280022d0c4401bcf3b0ea85870519",
    "userId": "558057c4256545bd8a307c37464003c9",
    "backendId": "5f2162cc64aeb90001d5bd3d",
    "backend": "aws-backend-file",
    "type": "cloudFS",
    "region": "ap-south-1",
    "availabilityZone": "ap-south-1a",
    "status": "creating",
    "tags": [
        {
            "key": "Name",
            "value": "ashitfs"
        }
    ],
    "metadata": {
        "fields": {
            "CreationTimeAtBackend": {
                "Kind": {
                    "StringValue": "2020-07-29T11:58:38Z"
                }
            },
            "CreationToken": {
                "Kind": {
                    "StringValue": "1wcZwO$MnD2!}j/K]p/9VliOKj%ty|^R^H$T"
                }
            },
            "FileSystemId": {
                "Kind": {
                    "StringValue": "fs-05cc5dd4"
                }
            },
            "FileSystemSize": {
                "Kind": {
                    "StructValue": {
                        "fields": {
                            "Timestamp": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "Value": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "ValueInIA": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "ValueInStandard": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            }
                        }
                    }
                }
            },
            "LifeCycleState": {
                "Kind": {
                    "StringValue": "creating"
                }
            },
            "Name": {
                "Kind": {
                    "StringValue": "ashitfs"
                }
            },
            "NumberOfMountTargets": {
                "Kind": {
                    "NumberValue": 0
                }
            },
            "OwnerId": {
                "Kind": {
                    "StringValue": "676599967111"
                }
            },
            "PerformanceMode": {
                "Kind": {
                    "StringValue": "generalPurpose"
                }
            },
            "ThroughputMode": {
                "Kind": {
                    "StringValue": "bursting"
                }
            }
        }
    }
}
```

----------------------------------------------------------------------------------------------------------------------------------

Example 2: Creating File Share for AWS Backend

POST http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares

Request:

```
{
    "name": "himanshufs",
    "description": "AWS FileShare",
    "backendId": "5f2162cc64aeb90001d5bd3d",
    "availabilityZone": "ap-south-1a",
    "type": "cloudFS",
    "tags": [
      {
        "key": "Name",
        "value": "himanshufs"
      }
    ],
    "metadata": {
      "PerformanceMode": "maxIO",
      "ThroughputMode": "bursting"
    }
}
```
Response:
```
{
    "id": "5f216581cb68c700018b8626",
    "createdAt": "2020-07-29T17:33:13",
    "updatedAt": "2020-07-29T17:33:13",
    "name": "himanshufs",
    "description": "AWS FileShare",
    "tenantId": "94b280022d0c4401bcf3b0ea85870519",
    "userId": "558057c4256545bd8a307c37464003c9",
    "backendId": "5f2162cc64aeb90001d5bd3d",
    "backend": "aws-backend-file",
    "type": "cloudFS",
    "region": "ap-south-1",
    "availabilityZone": "ap-south-1a",
    "status": "creating",
    "tags": [
        {
            "key": "Name",
            "value": "himanshufs"
        }
    ],
    "metadata": {
        "fields": {
            "CreationTimeAtBackend": {
                "Kind": {
                    "StringValue": "2020-07-29T12:03:13Z"
                }
            },
            "CreationToken": {
                "Kind": {
                    "StringValue": "ofoH]y?ph3LiJM8SiDDc|~7N$%)~{$#z8+^H"
                }
            },
            "FileSystemId": {
                "Kind": {
                    "StringValue": "fs-d7cf5e06"
                }
            },
            "FileSystemSize": {
                "Kind": {
                    "StructValue": {
                        "fields": {
                            "Timestamp": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "Value": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "ValueInIA": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "ValueInStandard": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            }
                        }
                    }
                }
            },
            "LifeCycleState": {
                "Kind": {
                    "StringValue": "creating"
                }
            },
            "Name": {
                "Kind": {
                    "StringValue": "himanshufs"
                }
            },
            "NumberOfMountTargets": {
                "Kind": {
                    "NumberValue": 0
                }
            },
            "OwnerId": {
                "Kind": {
                    "StringValue": "676599967111"
                }
            },
            "PerformanceMode": {
                "Kind": {
                    "StringValue": "maxIO"
                }
            },
            "ThroughputMode": {
                "Kind": {
                    "StringValue": "bursting"
                }
            }
        }
    }
}
```
----------------------------------------------------------------------------------------------------------------------------------
Example 3: Creating File Share for AWS Backend

POST http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares

Request:

```
{
    "name": "sanilfs",
    "description": "AWS FileShare",
    "backendId": "5f2162cc64aeb90001d5bd3d",
    "availabilityZone": "ap-south-1a",
    "type": "cloudFS",
    "encrypted": true,
    "encryptionSettings": {
        "KmsKeyId": "arn:aws:kms:ap-south-1:676599967111:key/a6e0716e-0e90-406b-b404-b8ee8de891fc"
    },
    "tags": [
        {
            "key": "Name",
            "value": "sanilfs"
        },
         {
            "key": "Organization",
            "value": "SODAFoundation"
        }
    ],
    "metadata": {
        "PerformanceMode": "maxIO",
        "ThroughputMode": "provisioned",
        "ProvisionedThroughputInMibps": 1
    }
}
```
Response:
```
{
    "id": "5f2165decb68c700018b8627",
    "createdAt": "2020-07-29T17:34:46",
    "updatedAt": "2020-07-29T17:34:46",
    "name": "sanilfs",
    "description": "AWS FileShare",
    "tenantId": "94b280022d0c4401bcf3b0ea85870519",
    "userId": "558057c4256545bd8a307c37464003c9",
    "backendId": "5f2162cc64aeb90001d5bd3d",
    "backend": "aws-backend-file",
    "type": "cloudFS",
    "region": "ap-south-1",
    "availabilityZone": "ap-south-1a",
    "status": "creating",
    "tags": [
        {
            "key": "Name",
            "value": "sanilfs"
        },
        {
            "key": "Organization",
            "value": "SODAFoundation"
        }
    ],
    "encrypted": true,
    "encryptionSettings": {
        "KmsKeyId": "arn:aws:kms:ap-south-1:676599967111:key/a6e0716e-0e90-406b-b404-b8ee8de891fc"
    },
    "metadata": {
        "fields": {
            "CreationTimeAtBackend": {
                "Kind": {
                    "StringValue": "2020-07-29T12:04:46Z"
                }
            },
            "CreationToken": {
                "Kind": {
                    "StringValue": "RjH5F9th@CYA4]Kpt+bqxX]911$4T)8)D2sS"
                }
            },
            "FileSystemId": {
                "Kind": {
                    "StringValue": "fs-cacf5e1b"
                }
            },
            "FileSystemSize": {
                "Kind": {
                    "StructValue": {
                        "fields": {
                            "Timestamp": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "Value": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "ValueInIA": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "ValueInStandard": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            }
                        }
                    }
                }
            },
            "LifeCycleState": {
                "Kind": {
                    "StringValue": "creating"
                }
            },
            "Name": {
                "Kind": {
                    "StringValue": "sanilfs"
                }
            },
            "NumberOfMountTargets": {
                "Kind": {
                    "NumberValue": 0
                }
            },
            "OwnerId": {
                "Kind": {
                    "StringValue": "676599967111"
                }
            },
            "PerformanceMode": {
                "Kind": {
                    "StringValue": "maxIO"
                }
            },
            "ProvisionedThroughputInMibps": {
                "Kind": {
                    "NumberValue": 1
                }
            },
            "ThroughputMode": {
                "Kind": {
                    "StringValue": "provisioned"
                }
            }
        }
    }
}
```
----------------------------------------------------------------------------------------------------------------------------------
AWS FS

Example 4: Creating File Share for Azure Backend

POST http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares

Request:

```
{
    "name": "anvithfs",
    "description": "Azure FileShare",
    "backendId": "5f115712125aa200016232db",
    "type": "cloudFS",
    "size": 1073741824,
    "metadata": {
        "createdby": "himanshu varshney",
        "organization": "SODAFoundation",
        "email": "himanshuvar@gmail.com"
    }
}
```
Response:
```
{
    "id": "5f2169ddcb68c700018b8628",
    "createdAt": "2020-07-29T17:51:49",
    "updatedAt": "2020-07-29T17:51:49",
    "name": "anvithfs",
    "description": "Azure FileShare",
    "tenantId": "94b280022d0c4401bcf3b0ea85870519",
    "userId": "558057c4256545bd8a307c37464003c9",
    "backendId": "5f21624864aeb90001d5bd3c",
    "backend": "azure-backend-file",
    "size": 1073741824,
    "type": "cloudFS",
    "region": "East Asia",
    "status": "creating",
    "metadata": {
        "fields": {
            "Content-Length": {
                "Kind": {
                    "NumberValue": 0
                }
            },
            "Date": {
                "Kind": {
                    "StringValue": "Wed, 29 Jul 2020 12:21:49 GMT"
                }
            },
            "Etag": {
                "Kind": {
                    "StringValue": "0x8D833B9F7D3A8BC"
                }
            },
            "Last-Modified": {
                "Kind": {
                    "StringValue": "Wed, 29 Jul 2020 12:21:49 GMT"
                }
            },
            "Server": {
                "Kind": {
                    "StringValue": "Windows-Azure-File/1.0 Microsoft-HTTPAPI/2.0"
                }
            },
            "X-Ms-Client-Request-Id": {
                "Kind": {
                    "StringValue": "4f051f59-ccd0-40f3-4807-e6346e64739e"
                }
            },
            "X-Ms-Request-Id": {
                "Kind": {
                    "StringValue": "c30e2293-a01a-002a-02a2-656efc000000"
                }
            },
            "X-Ms-Version": {
                "Kind": {
                    "StringValue": "2019-02-02"
                }
            }
        }
    }
}
```
----------------------------------------------------------------------------------------------------------------------------------
Azure FS

**List FileShare:**

GET http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares

Response:

```
{
    "fileshares": [
        {
            "id": "5f21646ecb68c700018b8625",
            "createdAt": "2020-07-29T17:28:38",
            "updatedAt": "2020-07-29T17:28:44",
            "name": "ashitfs",
            "description": "AWS FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5f2162cc64aeb90001d5bd3d",
            "backend": "aws-backend-file",
            "size": 6144,
            "type": "cloudFS",
            "region": "ap-south-1",
            "availabilityZone": "ap-south-1a",
            "status": "available",
            "tags": [
                {
                    "key": "Name",
                    "value": "ashitfs"
                }
            ],
            "metadata": {
                "fields": {
                    "CreationTimeAtBackend": {
                        "Kind": {
                            "StringValue": "2020-07-29T11:58:38Z"
                        }
                    },
                    "CreationToken": {
                        "Kind": {
                            "StringValue": "1wcZwO$MnD2!}j/K]p/9VliOKj%ty|^R^H$T"
                        }
                    },
                    "FileSystemId": {
                        "Kind": {
                            "StringValue": "fs-05cc5dd4"
                        }
                    },
                    "FileSystemSize": {
                        "Kind": {
                            "StructValue": {
                                "fields": {
                                    "Timestamp": {
                                        "Kind": {
                                            "NumberValue": 0
                                        }
                                    },
                                    "Value": {
                                        "Kind": {
                                            "NumberValue": 6144
                                        }
                                    },
                                    "ValueInIA": {
                                        "Kind": {
                                            "NumberValue": 0
                                        }
                                    },
                                    "ValueInStandard": {
                                        "Kind": {
                                            "NumberValue": 6144
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "LifeCycleState": {
                        "Kind": {
                            "StringValue": "available"
                        }
                    },
                    "Name": {
                        "Kind": {
                            "StringValue": "ashitfs"
                        }
                    },
                    "NumberOfMountTargets": {
                        "Kind": {
                            "NumberValue": 0
                        }
                    },
                    "OwnerId": {
                        "Kind": {
                            "StringValue": "676599967111"
                        }
                    },
                    "PerformanceMode": {
                        "Kind": {
                            "StringValue": "generalPurpose"
                        }
                    },
                    "ThroughputMode": {
                        "Kind": {
                            "StringValue": "bursting"
                        }
                    }
                }
            }
        },
        {
            "id": "5f216581cb68c700018b8626",
            "createdAt": "2020-07-29T17:33:13",
            "updatedAt": "2020-07-29T17:46:40",
            "name": "himanshufs",
            "description": "AWS FileShare Updated",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5f2162cc64aeb90001d5bd3d",
            "backend": "aws-backend-file",
            "size": 6144,
            "type": "cloudFS",
            "region": "ap-south-1",
            "availabilityZone": "ap-south-1a",
            "status": "available",
            "tags": [
                {
                    "key": "Name",
                    "value": "himanshufs"
                },
                {
                    "key": "Organization",
                    "value": "SODAFoundation"
                }
            ],
            "metadata": {
                "fields": {
                    "CreationTimeAtBackend": {
                        "Kind": {
                            "StringValue": "2020-07-29T12:03:13Z"
                        }
                    },
                    "CreationToken": {
                        "Kind": {
                            "StringValue": "ofoH]y?ph3LiJM8SiDDc|~7N$%)~{$#z8+^H"
                        }
                    },
                    "FileSystemId": {
                        "Kind": {
                            "StringValue": "fs-d7cf5e06"
                        }
                    },
                    "FileSystemSize": {
                        "Kind": {
                            "StructValue": {
                                "fields": {
                                    "Timestamp": {
                                        "Kind": {
                                            "NumberValue": 0
                                        }
                                    },
                                    "Value": {
                                        "Kind": {
                                            "NumberValue": 6144
                                        }
                                    },
                                    "ValueInIA": {
                                        "Kind": {
                                            "NumberValue": 0
                                        }
                                    },
                                    "ValueInStandard": {
                                        "Kind": {
                                            "NumberValue": 6144
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "LifeCycleState": {
                        "Kind": {
                            "StringValue": "available"
                        }
                    },
                    "Name": {
                        "Kind": {
                            "StringValue": "himanshufs"
                        }
                    },
                    "NumberOfMountTargets": {
                        "Kind": {
                            "NumberValue": 0
                        }
                    },
                    "OwnerId": {
                        "Kind": {
                            "StringValue": "676599967111"
                        }
                    },
                    "PerformanceMode": {
                        "Kind": {
                            "StringValue": "maxIO"
                        }
                    },
                    "ProvisionedThroughputInMibps": {
                        "Kind": {
                            "NumberValue": 1
                        }
                    },
                    "ThroughputMode": {
                        "Kind": {
                            "StringValue": "provisioned"
                        }
                    }
                }
            }
        },
        {
            "id": "5f2165decb68c700018b8627",
            "createdAt": "2020-07-29T17:34:46",
            "updatedAt": "2020-07-29T17:34:52",
            "name": "sanilfs",
            "description": "AWS FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5f2162cc64aeb90001d5bd3d",
            "backend": "aws-backend-file",
            "size": 6144,
            "type": "cloudFS",
            "region": "ap-south-1",
            "availabilityZone": "ap-south-1a",
            "status": "available",
            "tags": [
                {
                    "key": "Name",
                    "value": "sanilfs"
                },
                {
                    "key": "Organization",
                    "value": "SODAFoundation"
                }
            ],
            "encrypted": true,
            "encryptionSettings": {
                "KmsKeyId": "arn:aws:kms:ap-south-1:676599967111:key/a6e0716e-0e90-406b-b404-b8ee8de891fc"
            },
            "metadata": {
                "fields": {
                    "CreationTimeAtBackend": {
                        "Kind": {
                            "StringValue": "2020-07-29T12:04:46Z"
                        }
                    },
                    "CreationToken": {
                        "Kind": {
                            "StringValue": "RjH5F9th@CYA4]Kpt+bqxX]911$4T)8)D2sS"
                        }
                    },
                    "FileSystemId": {
                        "Kind": {
                            "StringValue": "fs-cacf5e1b"
                        }
                    },
                    "FileSystemSize": {
                        "Kind": {
                            "StructValue": {
                                "fields": {
                                    "Timestamp": {
                                        "Kind": {
                                            "NumberValue": 0
                                        }
                                    },
                                    "Value": {
                                        "Kind": {
                                            "NumberValue": 6144
                                        }
                                    },
                                    "ValueInIA": {
                                        "Kind": {
                                            "NumberValue": 0
                                        }
                                    },
                                    "ValueInStandard": {
                                        "Kind": {
                                            "NumberValue": 6144
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "LifeCycleState": {
                        "Kind": {
                            "StringValue": "available"
                        }
                    },
                    "Name": {
                        "Kind": {
                            "StringValue": "sanilfs"
                        }
                    },
                    "NumberOfMountTargets": {
                        "Kind": {
                            "NumberValue": 0
                        }
                    },
                    "OwnerId": {
                        "Kind": {
                            "StringValue": "676599967111"
                        }
                    },
                    "PerformanceMode": {
                        "Kind": {
                            "StringValue": "maxIO"
                        }
                    },
                    "ProvisionedThroughputInMibps": {
                        "Kind": {
                            "NumberValue": 1
                        }
                    },
                    "ThroughputMode": {
                        "Kind": {
                            "StringValue": "provisioned"
                        }
                    }
                }
            }
        },
        {
            "id": "5f2169ddcb68c700018b8628",
            "createdAt": "2020-07-29T17:51:49",
            "updatedAt": "2020-07-29T17:51:55",
            "name": "anvithfs",
            "description": "Azure FileShare",
            "tenantId": "94b280022d0c4401bcf3b0ea85870519",
            "userId": "558057c4256545bd8a307c37464003c9",
            "backendId": "5f21624864aeb90001d5bd3c",
            "backend": "azure-backend-file",
            "size": 1073741824,
            "type": "cloudFS",
            "region": "East Asia",
            "status": "available",
            "metadata": {
                "fields": {
                    "Content-Length": {
                        "Kind": {
                            "NumberValue": 0
                        }
                    },
                    "Date": {
                        "Kind": {
                            "StringValue": "Wed, 29 Jul 2020 12:21:55 GMT"
                        }
                    },
                    "Etag": {
                        "Kind": {
                            "StringValue": "0x8D833B9F7D3A8BC"
                        }
                    },
                    "Last-Modified": {
                        "Kind": {
                            "StringValue": "Wed, 29 Jul 2020 12:21:49 GMT"
                        }
                    },
                    "Server": {
                        "Kind": {
                            "StringValue": "Windows-Azure-File/1.0 Microsoft-HTTPAPI/2.0"
                        }
                    },
                    "Share-Usage-Bytes": {
                        "Kind": {
                            "NumberValue": 0
                        }
                    },
                    "X-Ms-Client-Request-Id": {
                        "Kind": {
                            "StringValue": "75d0c85e-5554-40f3-69ac-42a67cfce828"
                        }
                    },
                    "X-Ms-Has-Immutability-Policy": {
                        "Kind": {
                            "StringValue": "false"
                        }
                    },
                    "X-Ms-Has-Legal-Hold": {
                        "Kind": {
                            "StringValue": "false"
                        }
                    },
                    "X-Ms-Meta-Createdby": {
                        "Kind": {
                            "StringValue": "himanshu varshney"
                        }
                    },
                    "X-Ms-Meta-Email": {
                        "Kind": {
                            "StringValue": "himanshuvar@gmail.com"
                        }
                    },
                    "X-Ms-Meta-Organization": {
                        "Kind": {
                            "StringValue": "SODAFoundation"
                        }
                    },
                    "X-Ms-Request-Id": {
                        "Kind": {
                            "StringValue": "c30e229f-a01a-002a-06a2-656efc000000"
                        }
                    },
                    "X-Ms-Share-Quota": {
                        "Kind": {
                            "NumberValue": 1
                        }
                    },
                    "X-Ms-Version": {
                        "Kind": {
                            "StringValue": "2019-02-02"
                        }
                    }
                }
            }
        }
    ],
    "next": 4
}
```
---------------------------------------------------------------------------------------------------------------------------------
**Update FileShare:**

Update FileShare for AWS Backend:

PUT http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares/5f216581cb68c700018b8626

Request:

```
{
    "description": "AWS FileShare Updated",
    "tags": [
         {
            "key": "Organization",
            "value": "SODAFoundation"
        }
    ],
    "metadata": {
        "ThroughputMode": "provisioned",
        "ProvisionedThroughputInMibps": 1
    }
}
```
Response:
```
{
    "id": "5f216581cb68c700018b8626",
    "createdAt": "2020-07-29T17:33:13",
    "updatedAt": "2020-07-29T17:46:34",
    "name": "himanshufs",
    "description": "AWS FileShare Updated",
    "tenantId": "94b280022d0c4401bcf3b0ea85870519",
    "userId": "558057c4256545bd8a307c37464003c9",
    "backendId": "5f2162cc64aeb90001d5bd3d",
    "backend": "aws-backend-file",
    "size": 6144,
    "type": "cloudFS",
    "region": "ap-south-1",
    "availabilityZone": "ap-south-1a",
    "status": "updating",
    "tags": [
        {
            "key": "Name",
            "value": "himanshufs"
        },
        {
            "key": "Organization",
            "value": "SODAFoundation"
        }
    ],
    "metadata": {
        "fields": {
            "CreationTimeAtBackend": {
                "Kind": {
                    "StringValue": "2020-07-29T12:03:13Z"
                }
            },
            "CreationToken": {
                "Kind": {
                    "StringValue": "ofoH]y?ph3LiJM8SiDDc|~7N$%)~{$#z8+^H"
                }
            },
            "FileSystemId": {
                "Kind": {
                    "StringValue": "fs-d7cf5e06"
                }
            },
            "FileSystemSize": {
                "Kind": {
                    "StructValue": {
                        "fields": {
                            "Timestamp": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "Value": {
                                "Kind": {
                                    "NumberValue": 6144
                                }
                            },
                            "ValueInIA": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "ValueInStandard": {
                                "Kind": {
                                    "NumberValue": 6144
                                }
                            }
                        }
                    }
                }
            },
            "LifeCycleState": {
                "Kind": {
                    "StringValue": "updating"
                }
            },
            "Name": {
                "Kind": {
                    "StringValue": "himanshufs"
                }
            },
            "NumberOfMountTargets": {
                "Kind": {
                    "NumberValue": 0
                }
            },
            "OwnerId": {
                "Kind": {
                    "StringValue": "676599967111"
                }
            },
            "PerformanceMode": {
                "Kind": {
                    "StringValue": "maxIO"
                }
            },
            "ThroughputMode": {
                "Kind": {
                    "StringValue": "bursting"
                }
            }
        }
    }
}
```
---------------------------------------------------------------------------------------------------------------------------------
AWS FS Updated

Update FileShare for Azure Backend:

PUT http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares/5f2169ddcb68c700018b8628
Request:

```
{
  "description": "Azure FileShare Updated",
  "size": 2147483648
}
```
Response:
```
{
    "id": "5f2169ddcb68c700018b8628",
    "createdAt": "2020-07-29T17:51:49",
    "updatedAt": "2020-07-29T18:02:39",
    "name": "anvithfs",
    "description": "Azure FileShare Updated",
    "tenantId": "94b280022d0c4401bcf3b0ea85870519",
    "userId": "558057c4256545bd8a307c37464003c9",
    "backendId": "5f21624864aeb90001d5bd3c",
    "backend": "azure-backend-file",
    "size": 2147483648,
    "type": "cloudFS",
    "region": "East Asia",
    "status": "updating",
    "metadata": {
        "fields": {
            "Content-Length": {
                "Kind": {
                    "NumberValue": 0
                }
            },
            "Date": {
                "Kind": {
                    "StringValue": "Wed, 29 Jul 2020 12:32:39 GMT"
                }
            },
            "Etag": {
                "Kind": {
                    "StringValue": "0x8D833BB7B5508BD"
                }
            },
            "Last-Modified": {
                "Kind": {
                    "StringValue": "Wed, 29 Jul 2020 12:32:39 GMT"
                }
            },
            "Server": {
                "Kind": {
                    "StringValue": "Windows-Azure-File/1.0 Microsoft-HTTPAPI/2.0"
                }
            },
            "Share-Usage-Bytes": {
                "Kind": {
                    "NumberValue": 0
                }
            },
            "X-Ms-Client-Request-Id": {
                "Kind": {
                    "StringValue": "d1d8512e-cf2a-4b61-70c1-14401cb0449d"
                }
            },
            "X-Ms-Has-Immutability-Policy": {
                "Kind": {
                    "StringValue": "false"
                }
            },
            "X-Ms-Has-Legal-Hold": {
                "Kind": {
                    "StringValue": "false"
                }
            },
            "X-Ms-Meta-Createdby": {
                "Kind": {
                    "StringValue": "himanshu varshney"
                }
            },
            "X-Ms-Meta-Email": {
                "Kind": {
                    "StringValue": "himanshuvar@gmail.com"
                }
            },
            "X-Ms-Meta-Organization": {
                "Kind": {
                    "StringValue": "SODAFoundation"
                }
            },
            "X-Ms-Request-Id": {
                "Kind": {
                    "StringValue": "292bbac4-401a-008a-59a4-65a19a000000"
                }
            },
            "X-Ms-Share-Quota": {
                "Kind": {
                    "NumberValue": 1
                }
            },
            "X-Ms-Version": {
                "Kind": {
                    "StringValue": "2019-02-02"
                }
            }
        }
    }
}
```
---------------------------------------------------------------------------------------------------------------------------------
**Get FileShare:**

GET http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares/5f2165decb68c700018b8627

Response:

```
{
    "id": "5f2165decb68c700018b8627",
    "createdAt": "2020-07-29T17:34:46",
    "updatedAt": "2020-07-29T17:34:52",
    "name": "sanilfs",
    "description": "AWS FileShare",
    "tenantId": "94b280022d0c4401bcf3b0ea85870519",
    "userId": "558057c4256545bd8a307c37464003c9",
    "backendId": "5f2162cc64aeb90001d5bd3d",
    "backend": "aws-backend-file",
    "size": 6144,
    "type": "cloudFS",
    "region": "ap-south-1",
    "availabilityZone": "ap-south-1a",
    "status": "available",
    "tags": [
        {
            "key": "Name",
            "value": "sanilfs"
        },
        {
            "key": "Organization",
            "value": "SODAFoundation"
        }
    ],
    "encrypted": true,
    "encryptionSettings": {
        "KmsKeyId": "arn:aws:kms:ap-south-1:676599967111:key/a6e0716e-0e90-406b-b404-b8ee8de891fc"
    },
    "metadata": {
        "fields": {
            "CreationTimeAtBackend": {
                "Kind": {
                    "StringValue": "2020-07-29T12:04:46Z"
                }
            },
            "CreationToken": {
                "Kind": {
                    "StringValue": "RjH5F9th@CYA4]Kpt+bqxX]911$4T)8)D2sS"
                }
            },
            "FileSystemId": {
                "Kind": {
                    "StringValue": "fs-cacf5e1b"
                }
            },
            "FileSystemSize": {
                "Kind": {
                    "StructValue": {
                        "fields": {
                            "Timestamp": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "Value": {
                                "Kind": {
                                    "NumberValue": 6144
                                }
                            },
                            "ValueInIA": {
                                "Kind": {
                                    "NumberValue": 0
                                }
                            },
                            "ValueInStandard": {
                                "Kind": {
                                    "NumberValue": 6144
                                }
                            }
                        }
                    }
                }
            },
            "LifeCycleState": {
                "Kind": {
                    "StringValue": "available"
                }
            },
            "Name": {
                "Kind": {
                    "StringValue": "sanilfs"
                }
            },
            "NumberOfMountTargets": {
                "Kind": {
                    "NumberValue": 0
                }
            },
            "OwnerId": {
                "Kind": {
                    "StringValue": "676599967111"
                }
            },
            "PerformanceMode": {
                "Kind": {
                    "StringValue": "maxIO"
                }
            },
            "ProvisionedThroughputInMibps": {
                "Kind": {
                    "NumberValue": 1
                }
            },
            "ThroughputMode": {
                "Kind": {
                    "StringValue": "provisioned"
                }
            }
        }
    }
}
```
---------------------------------------------------------------------------------------------------------------------------------
Azure FS Updated

**Delete FileShare:**

DELETE http://192.168.20.162:8089/v1/94b280022d0c4401bcf3b0ea85870519/file/shares/5f21646ecb68c700018b8625

Response:

{}
**Special notes for your reviewer**:

### CODE REVIEW

File: file/pkg/service/service.go [hunk 1]
```diff
@@ -251,6 +234,7 @@ func (f *fileService) CreateFileShare(ctx context.Context, in *pb.CreateFileShar
 	log.Debugf("Create File share response, fileshare: %+v\n", out.Fileshare)
 
 	log.Info("Created file share successfully.")
+	go f.SyncFileShare(ctx, out.Fileshare, backend.Backend)
```
Comment (kumarashit) [inline file/pkg/service/service.go:237]: Please put a comment for why this is required

Comment (himanshuvar) [reply file/pkg/service/service.go:239]: Done

**Review by NajmudheenCT — APPROVED** [review]
LGTM

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**